commit 3807f6133efb54380787dcc5f3a8b4ef1ce30053
Author: Paul Pluzhnikov <ppluzhnikov@google.com>
Date:   Sat Jul 23 16:49:16 2011 +0000

    2011-07-23  Paul Pluzhnikov  <ppluzhnikov@google.com>
    
    	* elfread.c (elf_rel_plt_read): Fix off-by-one bug.

Index: gdb-7.2/gdb/elfread.c
===================================================================
--- gdb-7.2.orig/gdb/elfread.c
+++ gdb-7.2/gdb/elfread.c
@@ -669,13 +669,13 @@ elf_rel_plt_read (struct objfile *objfil
 	 OBJFILE the symbol is undefined and the objfile having NAME defined
 	 may not yet have been loaded.  */
 
-      if (string_buffer_size < name_len + suffix_len)
+      if (string_buffer_size < name_len + suffix_len + 1)
 	{
 	  string_buffer_size = 2 * (name_len + suffix_len);
 	  string_buffer = xrealloc (string_buffer, string_buffer_size);
 	}
       memcpy (string_buffer, name, name_len);
-      memcpy (&string_buffer[name_len], suffix, suffix_len);
+      memcpy (&string_buffer[name_len], suffix, suffix_len + 1);
 
       msym = record_minimal_symbol (string_buffer, name_len + suffix_len, 1,
 				    address, mst_slot_got_plt, got_plt,
